### ISSUE TITLE
Crash in AWSDataStorePlugin/AWSDataStorePlugin+DataStoreBaseBehavior.swift

### ISSUE THREAD

**Issue by fzy-spyro** [aws-amplify/amplify-swift#3198]
### Describe the bug

AWSDataStorePlugin Crashes. 

We are observing crashes similar to the one reported previously here: https://github.com/aws-amplify/amplify-swift/issues/2841

We have observed that crash happens when using `save` and `query` operations. 

Sample error reported: 

`AWSDataStorePlugin/AWSDataStorePlugin+DataStoreBaseBehavior.swift:68: Fatal error: Unexpectedly found nil while implicitly unwrapping an Optional value`

Sample Stacktraces:

```
0   libswiftCore.dylib              0x331593ed0         _assertionFailure
1   libswiftCore.dylib              0x331593ecc         _assertionFailure
2   libswiftCore.dylib              0x331593d64         _assertionFailure
3   libswiftCore.dylib              0x3315936f0         _assertionFailure
4   MyApp                       0x200893890         AWSDataStorePlugin.save<T> (AWSDataStorePlugin+DataStoreBaseBehavior.swift:68)
5   MyApp                       0x2008967e8         AWSDataStorePlugin.save<T> (AWSDataStorePlugin+DataStoreBaseBehavior.swift:79)
6   libswift_Concurrency.dylib      0x3521fc8cc         withCheckedContinuation<T>
7   libswift_Concurrency.dylib      0x3521fc918         withCheckedContinuation<T>
8   libswift_Concurrency.dylib      0x3521fc9f4         withUnsafeThrowingContinuation<T>
9   libswift_Concurrency.dylib      0x352237944         swift::runJobInEstablishedExecutorContext
```

```
0   MyApp                      0x203076380         [inlined] value
1   MyApp                       0x203076380         AWSDataStorePlugin.query<T> (AWSDataStorePlugin+DataStoreBaseBehavior.swift:216)
2   MyApp                       0x2030762cc         AWSDataStorePlugin.query<T>
3   MyApp                       0x20307c8b8         [inlined] AWSDataStorePlugin.query<T> (AWSDataStorePlugin+DataStoreBaseBehavior.swift:231)
4   MyApp                       0x20307c8b8         AWSDataStorePlugin.query<T>
5   libswift_Concurrency.dylib      0x32e3be8cc         withCheckedContinuation<T>
6   libswift_Concurrency.dylib      0x32e3be918         withCheckedContinuation<T>
7   libswift_Concurrency.dylib      0x32e3be9f4         withUnsafeThrowingContinuation<T>
8   libswift_Concurrency.dylib      0x32e3f9944         swift::runJobInEstablishedExecutorContext
```

Steps

### Steps To Reproduce

```swift
No steps to reporduce. Crash happens only sometimes. 

We have been able to pinpoint this to the time when DataStore starts/stops and app tries to query/save data "in close proximity" to starting stopping. 

Eg case: app and datastore starts and we try to save push notification token to datastore
```

### Expected behavior

No crash

### Amplify Framework Version

2.11.5

### Amplify Categories

DataStore

### Dependency manager

Swift PM

### Swift version

5.7

### CLI version

12.2.3

### Xcode version

14.2, 14.3.1

### Relevant log output

```shell
N/A
```

### Is this a regression?

Yes

### Platforms

iOS

### OS Version

all

### Device

all 

### Specific to simulators

no

**Comment by 5d** [comment]
Thanks for opening the issue @fzy-spyro 
We will investigate and get back to you soon.

**Comment by zachfields** [comment]
I am having this same issue with tvOS.

It runs perfectly fine on the tvOS simulator, but when run on an actual device (Apple TV 4K, tvOS 16.6), it crashes when a query is run.

**Comment by fzy-spyro** [comment]
Hey, 
I see the PR with a fix is merged alrady. Do you have any news which release it will be in? 
Thanks in advance

**Comment by 5d** [comment]
Hi @fzy-spyro , we are planning to have a release this week.

**Comment by fzy-spyro** [comment]
@5d awesome, thanks

**Comment by atierian** [comment]
The fix for this was released in [Amplify Swift 2.17.2](https://github.com/aws-amplify/amplify-swift/releases/tag/2.17.2)
Thanks again for reporting this!
